### ISSUE TITLE
Host and Port

### ISSUE THREAD

**Issue by StefanoGiu** [home-assistant/home-assistant.github.io#28889]
### Feedback

What is the host/port I need to use?

### URL

https://www.home-assistant.io/integrations/wyoming/

### Version

2023.9.1

### Additional information

What is the host/port I need to use?

**Comment by synesthesiam** [comment]
Under Home Assistant OS, this should be automatically discovered.
If you are [running the Docker images](https://github.com/rhasspy/wyoming-addons) instead, you will need to use the host/port of the running container(s).

**Comment by StefanoGiu** [comment]
I don't use neither HAOS nor docker...

**Comment by gsaslis** [comment]
@StefanoGiu I also ran into the same issue and this explanation helped: 
https://github.com/home-assistant/home-assistant.io/issues/27252#issuecomment-1534641592

**Comment by StefanoGiu** [comment]
I don't use containers on my PC... do I need container just for Wyoming ?

**Comment by gsaslis** [comment]
My answer was for the case where you are hosting piper/whisper as independent process (not as home assistant addons). If that is your scenario, you just need to fill in the wyoming protocol form twice - once for piper and once for whisper.

**Comment by StefanoGiu** [comment]
But it's not clear to me how to install Wyoming on Windows if I don't use HAOS and I don't use docker containers...

Is there a way to install it? Where is the documentation? I also installed Whisper custom component (not the add-on) and it's properly installed, but I don't see it coming up under the assistant STT dropdown menu...

**Comment by gsaslis** [comment]
as far as I understand it, you don't install wyoming. you install piper and whisper and then you configure the wyoming protocol integration in home assistant to connect to those running processes (via the port they open)

**Comment by frenck** [comment]
> But it's not clear to me how to install Wyoming

You don't install Wyoming. Wyoming is a protocol, that is used/provided by Whisper and Piper programs (but not limited too, everyone could create a compatible software program using this standard).

**Comment by StefanoGiu** [comment]
Guys, please check the situation of HA when you don't run it as HAOS or Docker...

If I add the integration Whisper or Piper, it tells me, with a popup, that I need to install Wyoming protocol, if I click ok, then it jumps to Host/Port popup and I don't know what to insert into it...

Has someone tested this scenario before?

It seems not...

**Comment by frenck** [comment]
> Guys, please check the situation of HA when you don't run it as HAOS or Docker...

That is a very advanced use case, mostly used by developers.

> If I add the integration Whisper or Piper, it tells me, with a popup, that I need to install Wyoming protocol, if I click ok, then it jumps to Host/Port popup and I don't know what to insert into it...

You put in the host and port of your Piper and Whisper instance, the ones you have set up yourself. You know what to fill in in that case, not us. These services and software are not provided by Home Assistant.

> Has someone tested this scenario before?

Sure. Many have. As a matter of fact, even when using the add-on, it uses the same logic and we have unit tests for to ensure it works.

This is becoming a discussion on knowledge. Please, don't take this the wrong way, but these things (like the host and ports of applications) are a mystery to you; you might want to consider running a different installation method. Again, I'm not saying this to offend you, but mean it really well. As the one you are running now, will require more in-depth knowledge of things like Linux administration and networking.

../Frenck

**Comment by StefanoGiu** [comment]
Frenck I don't think you are understanding or reading what I wrote... 

You say "You put in the host and port of your Piper and Whisper instance, the ones you have set up yourself. You know what to fill in in that case, not us. These services and software are not provided by Home Assistant."... but I'm asking how to install Piper and Whisper because --> As I said before <-- If I try to add Whisper of Piper from Integration menu on my HA that is neither running on HAOS nor in docker containers, it will jump on Wyoming popup where it asks me host and port...

Therefore I cannot configure Whisper or Piper if I don't first install "Wyoming Protocol" integration... and again... I don't have add-ons options... so why you talk about add-ons ?

I've been running HA since years... I have complex integrations in place, hundred of thousands of lines of YAML code, custom components that I developed by myself and even patch that I made for some standard HA components that are buggy... so I don't think I need to change installation method...

If someone... instead of writing here... would have tested this scenario before, they would have realized what I'm trying to explain...

**Comment by frenck** [comment]
> but I'm asking how to install Piper and Whisper because

Please consult the documentation of those applications. We don't provided documentation on how to install Philips Hue, MySQL, deCONZ, <insert any other hardware or application>, ... either.

> so why you talk about add-ons ?

Because that is our ecosystem, of which you only use a small part. That is up to you, but everything we do is aimed at the 95% of our users that do.

> If someone... instead of writing here... would have tested this scenario before, they would have realized what I'm trying to explain...

I did and I realize what you are trying to explain, yet, I don't think it should have been a question. If it is, there is a mismatch in what you use versus what you probably should have used.

../Frenck

**Comment by synesthesiam** [comment]
@StefanoGiu Whisper and Piper are libraries that need some kind of wrapper around them for HA. Wyoming is that wrapper (custom protocol over TCP), and we have prepared both add-ons and Docker images with libraries + wrapper.

On the HA side, the Wyoming integration communications via TCP with the wrapper(s). So it needs the host and port where the wrapper is running, which is either automatically set via discovery (add-on) or manually configured by the user (Docker).

Hope this helps.

**Comment by StefanoGiu** [comment]
@frenck ...  I now understand your ability not to reply questions when you don't know the answer... you just mess everything around... being foggy doesn't really help here... if you don't know the answer, just don't reply.

@synesthesiam I don't either use HAOS nor have docker for containerization installed and I don't want to install docker just for Wyoming. 

Any alternative?

**Comment by synesthesiam** [comment]
Please don't be rude, @StefanoGiu. @frenck is correct and knowledgeable about Wyoming/Whisper/Piper.  You are simply asking for something well outside of the supported use cases.

> Any alternative?

You can see how the Docker images are constructed here: https://github.com/rhasspy/wyoming-addons
The `wyoming-whisper` and `wyoming-piper` Python packages can be executed to run Wyoming servers. The `wyoming-piper` library needs an executable binary for [Piper](https://github.com/rhasspy/piper/) available too.

**Comment by L81ker** [comment]
since the last update (affecting AppDaemon), Piper has vanished from my list of voices and the Wyoming protocol now asks for host and port information, i.e. it appears to no longer be finding the mapping to the python packages. How do I resolve this? Do I need to reinstall Piper, Whisper etc?

**Comment by synesthesiam** [comment]
Yes, do an uninstall and reinstall of the affected add-ons. This is the only way I know to clear the discovery info.

**Comment by L81ker** [comment]
> Yes, do an uninstall and reinstall of the affected add-ons. This is the only way I know to clear the discovery info.

Ok, cheers

**Comment by yunyuyuan** [comment]
@StefanoGiu 
I understand your situation because I've encountered it too. My Home Assistant is installed on a NUC, but I haven't purchased an ESP microphone yet. Therefore, I want to process my speech on my Windows laptop and send voice commands to Home Assistant on the NUC. I found https://github.com/Purfview/whisper-standalone-win, which can run directly on Windows for speech recognition. However, I haven't found a way to integrate whisper-standalone-win into the Wyoming protocol. It seems like I need to write a socket program in Python to send the results of whisper-standalone-win to Home Assistant.

**Comment by yunyuyuan** [comment]
I have found this repo, https://github.com/rhasspy/wyoming-faster-whisper, which can run whisper with wyoming protocol supported, you can install(go to **Local Install** section) and run it on your machine:
* If you run it on the same machine which home-assistant installed, the `Host` field should be `127.0.0.1`.
* If you run it on the other machine (make sure in the same network with your home-assistant), the `Host` field should be its ip.
